From 1d6c17f9cd5fb666672adf1149a834c169023a25 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 27 May 2021 12:24:27 +0300
Subject: [PATCH 3/7] s32g: Move PMIC WDG programming after initializations

PMIC has to be reprogrammed after pinmuxing and clocks
initializations.

Upstream-Status: Pending

Issue: ALB-7139
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32g/s32g_resume.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/plat/nxp/s32g/s32g_resume.c b/plat/nxp/s32g/s32g_resume.c
index b8b647e34..7053c82e1 100644
--- a/plat/nxp/s32g/s32g_resume.c
+++ b/plat/nxp/s32g/s32g_resume.c
@@ -32,14 +32,14 @@ void s32g_resume_entrypoint(void)
 {
 	int ret;
 
-	ret = pmic_disable_wdg();
-	if (ret)
-		ERROR("Failed to disable VR5510 watchdog\n");
-
 	/* Prepare resume operation */
 	s32g_ncore_isol_cluster0();
 	s32g_early_plat_init(true);
 
+	ret = pmic_disable_wdg();
+	if (ret)
+		ERROR("Failed to disable VR5510 watchdog\n");
+
 	reset_rtc();
 	s32gen1_wkpu_reset();
 
-- 
2.17.1

